maxim/tests/test_connection_retry_logic.py

```
@@ -0,0 +1,526 @@
+            ConnectionError("Connection failed"),
+            ConnectTimeout("Connection timeout"),
+            ReadTimeout("Read timeout"),
+            ProtocolError("Protocol error"),
+            MaxRetryError(None, None, "Max retries exceeded"),
+        ]
+
+        for error in connection_errors:
+            with self.subTest(error=type(error).__name__):
+                with patch.object(
+                    self.maxim_api.connection_pool, "get_session"
+                ) as mock_session_context:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)